htlcswitch/switch_test.go

```diff
@@ -999,6 +1001,7 @@ func TestSwitchForwardFailAfterFullAdd(t *testing.T) {
 
 	cdb, err := channeldb.Open(tempPath)
 	require.NoError(t, err, "unable to open channeldb")
+	defer cdb.Close()
```

Comment:
Should we use `t.Cleanup()` for these as well?

Reply:
If the call is `defer`'d within the body of a `Test` function, then `t.Cleanup` is almost identical. The only difference is that callbacks passed to `t.Cleanup` aren't called if the test panics. 

I prefer `defer` myself, just for brevity.

Reply:
But [as noted here](https://stackoverflow.com/a/64178378), `t.Cleanup` is useful for cleanup of parallelized tests.

Happy to follow whichever convention you'd prefer.

Reply:
You're right, `defer` is shorter and also works when the test panics. But there's also https://github.com/lightningnetwork/lnd/pull/6864 which will move many of the `defer`s over to `t.Cleanup()`, so I think it would be nice to follow the same pattern here.

Reply:
Aye fair point! Rebased & updated to use `t.Cleanup`